docs/source/conf.py

```diff
@@ -12,8 +12,10 @@
 # All configuration values have a default; values that are commented out
 # serve to show the default.
 
-import sys
+from __future__ import annotations
```

Comment:
As far as I know this import is not required in 3.7 and only enables ["postponed evaluation"](https://docs.python.org/3/whatsnew/3.7.html#pep-563-postponed-evaluation-of-annotations) which is a feature of later versions.

The annotations you added do not make use of the postponed evaluation.

Reply:
Why resist importing from `__future__`...  It is part of the language.

$ `docker run --rm -it python:3.7 bash ` # Wait for that container to open...
# `apt-get -qq update && apt-get install -y python3 && python3 --version ` # Python 3.7.13...
# `python3.7 -c "d: dict | None = None" ` # TypeError: unsupported operand type(s) for |: 'type' and 'NoneType'